src/core/net/ip6.cpp

```diff
@@ -426,6 +427,11 @@ ThreadError Ip6::HandleDatagram(Message &message, Netif *netif, int8_t interface
     {
         if (netif != NULL)
         {
+            if (otGetDeviceRole() == kDeviceRoleChild)
```

Comment:
Yes, that makes sense now and agree we need to fix this.  I think this fix would work, but I just submitted #390 which exposes the forwarding capability from `Ip6` and removes any Thread Role-specific logic.  Please review when you get a chance.